server/src/main/java/org/elasticsearch/action/admin/indices/alias/IndicesAliasesRequest.java

```diff
@@ -301,6 +317,9 @@ public AliasActions index(String index) {
                 throw new IllegalArgumentException("[index] can't be empty string");
             }
             this.indices = new String[] { index };
+            if (this.originalIndices == null) {
+                this.originalIndices = this.indices;
+            }
```

Comment:
There is an `originalAliases` field which is created for the same purpose: to keep the aliases from the request before they are resolved by the security layer. Unfortunately, despite being analogous, `orginalIndices` and `orginalAliases` are handled slightly differently. This is because the `indices` is modified in the security layer by calling the `indices()` function a second time which updates the field; thus we must check if `originalIndices` is null to avoid setting it a second time. This differs from how aliases are set. They are set the first time (along with `originalAliases`) in the `aliases()` function, then only the `aliases` field (and not `originalAliases`) is updated in the `replaceAliases()` function.

Anyway, I'd prefer that this code behaved the same as the similar code for aliases, but I think making such a change would be a too far reaching change.

Reply:
Yikes, yeah if the request is reused this will not work correctly. I can't think of any way to store the unresolved value which will work with reuse of the request. (Well we could change the security layer's call from `indices()` to a new function `replaceIndices()` as is done for aliases, but that seems well out of scope).

Using security-substituted index names isn't horrible. Unfortunately, the index and alias names are not resolved in the request when it enters TransportIndicesAliasesAction if security is disabled. This class calls out to the resolver. To keep security and non-security behavior identical, some tracking of index/alias resolved names would need to be added to the already complicated main loop.

Do you think it would be crazy to just not include any information about the action in the response? And require the user correlate the result list with the request list? Joe correctly pointed out the that the user cannot be expected to still have the request around ... but no information might be better than inconsistent information.

Reply:
I think using the `originalIndices` behavior as it is in this PR is more confusing and possibly incorrect. For now I think it'd be better to use the security-substituted index name, and perhaps track the originals differently the same work as #107017, what do you think?